doc/reference/guidelines.rst

```diff
@@ -0,0 +1,48 @@
+.. highlight:: python
+
+=================
+Coding Guidelines
+=================
+
+Python
```

Comment:
Things to add maybe
 - computed fields in v8: `_compute_<field_name>`
 - defaults for fields: `_default_<field_name>`